fhevm/precompiles.go

```diff
@@ -176,19 +177,37 @@ func FheLibRun(environment EVMEnvironment, caller common.Address, addr common.Ad
 		logger.Error("fheLib precompile error", "err", err, "input", hex.EncodeToString(input))
 		return nil, err
 	}
+	otelCtx := environment.OtelContext()
+	tracer := otel.Tracer("fhevm")
 	// first 4 bytes are for the function signature
 	signature := binary.BigEndian.Uint32(input[0:4])
 	switch signature {
```

Comment:
I'm the author of the initial switch statement (looked good idea at the time for me) and maybe in the future we can replace this with a map of small abstraction, like structs of all the fhevm functions which contain, say
```
type fhevmPrecompileMethod {
  name string
  implementation <function signature which implements current switch block>
}
```
And have a generated map of these for first 4 bytes of method signature
```
map[[]byte]fhevmPrecompileMethod
```
Then we could just do one lookup to the map, and do one trace using name of the method not to repeat ourselves.

Also, in future bwCompatBytes logic could be just moved inside implementation method, say `fheAddRun`.
This was needed before when we supported separate precompile smart contracts instead of one, we no longer need that now

Reply:
Yes, this seems a nice refactor, I will do that just after we merge this (probably today)